### ISSUE TITLE
Removal of Non-redundant return

### ISSUE THREAD

**Issue by garricn** [nicklockwood/SwiftFormat#100]
I'm seeing an issue where a non-redundant `return` is being removed. The `return` is not removed when using `--disable redundantReturn` so I'm pretty sure it's that rule.

Here is a code snippet where its happening (Please forgive the ugly legacy code. Code has also been stripped of Farmers related context) :

```
func example() -> Bool {
    if let dictionary = response["string"] as? [String: AnyObject] {
        let array = dictionary["otherString"] as! [NSDictionary]
        if array.count > 0 {
            for element in array {
                if dictionary["anotherString"] as! String == "YES" || dictionary["someString"] as! String == "YES" {
                    return false
                }
            }
        }
        let otherDictionary = response["string"] as! [String: AnyObject]
        let otherArray = otherDictionary["otherString"] as! [NSDictionary]
        if otherArray.count > 0 {
            for element in otherArray {

                if otherDictionary["someString"] as! String == "YES" {
                    return false
                }
            }
        }
    }
    return true
}
```

Expected: zero return statements removed
Actual: first return statement removed

**Comment by garricn** [comment]
This was not happening on version 0.24. It only occurred after upgrading to 0.25.1.

**Comment by garricn** [comment]
I have another code example where its happening. I can get it to you later if you want it.

**Comment by nicklockwood** [comment]
@garricn yes, please do - it may be a separate bug.

**Comment by garricn** [comment]
Expected: no returns removed
Actual: first two returns removed

```
func example(_: [String: AnyObject]) -> Bool {
    let thing = thing[""] as! [String: AnyObject]
    let thing = thing[""] as! [NSDictionary]

    let thing = thing[""] as! [String: AnyObject]
    let thing = thing[""] as! [NSDictionary]

    let thing = thing[""] as! [String: AnyObject]
    let thing = thing[""] as! [NSDictionary]

    let thing = thing[""] as! [String: AnyObject]
    let thing = thing[""] as! [NSDictionary]

    for element in thing {
        for element in array {
            if thing[""] as! String == thing[""] as! String
                && thing[""] as! String == ""
                && thing[""] as! String == "" {
                return true
            }
        }
    }

    for element in array {
        for element in array {
            if thing[""] as! String == thing[""] as! String
                && thing[""] as! String == ""
                && thing[""] as! String == "" {
                return true
            }
        }
    }
    return false
}
```

**Comment by nicklockwood** [comment]
@garricn fixed in 0.25.2. Thanks!

**Comment by garricn** [comment]
Thank you!
